src/message.cpp

```diff
@@ -30,33 +30,41 @@ static const char *    g_warningStr = "warning: ";
 static const char *    g_errorStr = "error: ";
 static FILE *          g_warnFile = stderr;
 static WARN_AS_ERROR_t g_warnBehavior = WARN_AS_ERROR_t::NO;
+static QCString        g_warnlogFile;
+static bool            g_warnlogTemp = false;
 static std::atomic_bool g_warnStat = false;
 static std::mutex      g_mutex;
 
 void initWarningFormat()
 {
   g_warnFormat = Config_getString(WARN_FORMAT);
   g_warnLineFormat = Config_getString(WARN_LINE_FORMAT);
-  QCString logFile = Config_getString(WARN_LOGFILE);
+  g_warnBehavior = Config_getEnum(WARN_AS_ERROR);
+  g_warnlogFile = Config_getString(WARN_LOGFILE);
+  if (g_warnlogFile.isEmpty() && g_warnBehavior == WARN_AS_ERROR_t::FAIL_ON_WARNINGS_PRINT)
+  {
+    uint pid = Portable::pid();
+    g_warnlogFile.sprintf("doxygen_warnings_temp_%d.tmp",pid);
+    g_warnlogTemp = true;
+  }
 
-  if (!logFile.isEmpty())
+  if (!g_warnlogFile.isEmpty())
   {
-    if (logFile == "-")
+    if (g_warnlogFile == "-")
     {
       g_warnFile = stdout;
     }
-    else if (!(g_warnFile = Portable::fopen(logFile,"w")))
+    else if (!(g_warnFile = Portable::fopen(g_warnlogFile,"w")))
```

Comment:
Hey there, if this log file has a foldername in it, but the folder does not exist, this call is not creating the folder. Then, the log error message appears, and then my CI instance looking for the log output in that folder does not find it.